src/Provider/AbstractProvider.php

```diff
@@ -445,7 +445,9 @@ protected function getAccessTokenBody(array $params)
      */
     protected function getAccessTokenOptions(array $params)
     {
-        $options = [];
+        $options = [
+          'headers' => $this->getAccessTokenRequestHeaders($params),
+        ];
 
         if ($this->getAccessTokenMethod() === self::METHOD_POST) {
             $options['body'] = $this->getAccessTokenBody($params);
```

Comment:
This is the only condition that requires having headers. Sending `application/x-www-form-urlencoded` for a `METHOD_GET` would be wrong.